website/docs/developer/research/research-deep-glacier-migration.mdx

```diff
@@ -0,0 +1,242 @@
+
+The boto3 `copy_object` function can be used in the python lambda function as shown below. 
+
+```python
+import boto3
+
+s3 = boto3.client('s3')
+#this list should be provided by the user
+retrieve_objects = ['test1.xml', 'test2.xml']
+bucket_name = 'orca-rhassan-sandbox-glacier'
+for migrate_object in retrieve_objects:
+    try:
+        response = s3.copy_object(
+            Bucket= bucket_name,  #bucket name where the object is stored currently
+            CopySource={'Bucket': bucket_name, 'Key': migrate_object},
+            Key = migrate_object, #keyname of object after migration
+            StorageClass='DEEP_ARCHIVE'
+        )
+        print(response)
+        print(f"copy succeeded for object {migrate_object} having E tag " + response["CopyObjectResult"]["ETag"])
+    except Exception as ex:
+        print(ex)
+```
+The resultant output from the boto3 `copy_object` command can bee seen below.
+```
+{'ResponseMetadata': {'RequestId': 'EM3XDZ6BB7FJJ88W', 'HostId': '/HayQvl8E2cCsclvO0b4MQUHAJnf7exmJhCILDsXfCieuaJIrAt5ZD9ocOkaam6rQbgXt0qx18A=', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-id-2': '/HayQvl8E2cCsclvO0b4MQUHAJnf7exmJhCILDsXfCieuaJIrAt5ZD9ocOkaam6rQbgXt0qx18A=', 'x-amz-request-id': 'EM3XDZ6BB7FJJ88W', 'date': 'Sun, 24 Jul 2022 18:13:32 GMT', 'x-amz-storage-class': 'DEEP_ARCHIVE', 'content-type': 'application/xml', 'server': 'AmazonS3', 'content-length': '234'}, 'RetryAttempts': 0}, 'CopyObjectResult': {'ETag': '"66d8119b127d165f8bc66fa66e0e35a6"', 'LastModified': datetime.datetime(2022, 7, 24, 18, 13, 32, tzinfo=tzutc())}}
+copy succeeded for object test1.xml having E tag "66d8119b127d165f8bc66fa66e0e35a6"
+{'ResponseMetadata': {'RequestId': 'SA312G5B2RRN5P0C', 'HostId': '/YIl4aUR+CAov4/pxzzd8oVORmYYsewTbwZfnVDkWRa+8g/epeQ7KEZGqSVgcx5LheMhU/dXMfw=', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-id-2': '/YIl4aUR+CAov4/pxzzd8oVORmYYsewTbwZfnVDkWRa+8g/epeQ7KEZGqSVgcx5LheMhU/dXMfw=', 'x-amz-request-id': 'SA312G5B2RRN5P0C', 'date': 'Sun, 24 Jul 2022 18:13:33 GMT', 'x-amz-storage-class': 'DEEP_ARCHIVE', 'content-type': 'application/xml', 'server': 'AmazonS3', 'content-length': '234'}, 'RetryAttempts': 0}, 'CopyObjectResult': {'ETag': '"14694ab9e3089a7c29e0dc9add4a82ab"', 'LastModified': datetime.datetime(2022, 7, 24, 18, 13, 33, tzinfo=tzutc())}}
+copy succeeded for object test2.xml having E tag "14694ab9e3089a7c29e0dc9add4a82ab"
+```
+
+A detailed sequence diagram that goes into more depth on the steps and actions needed for migrating data to DEEP ARCHIVE is shown below. The sequence diagram uses the recommended steps and approaches called out above.
+
+<MyImage
+    imageSource={useBaseUrl('img/Deep-Archive-Migration-workflow-specific-files.svg')}
```

Comment:
`Migration Copy` does not appear to be triggered in any way.

Reply:
this is shown in the architecture drawing. The migration report is triggered by an API gateway endpoint
See ORCA-Deep-Archive-Migration-Architecture-specific-files.svg

Reply:
I'm just talking about the sequence diagram. Presently, `Migration Copy` has no trigger that tells it to copy files to Deep Archive storage (line 12) and the `Migration Report` is reading from the queue, but not doing anything with that information (line 11).